### ISSUE TITLE
Reconsider design of restart events issued when rolling Kafka Pods

### ISSUE THREAD

**Issue by scholzj** [strimzi/strimzi-kafka-operator#10958]
Currently, when the Kafka pods are rolled, we issue Kubernetes Events describing the reason for the restart. It is done only for the Kafka, Connect and MM2 node restarts. The events are issued to the Pods as the main objects.

This approach has several issues:
* Pods have many events when they are restarted. So the restart reason event is issued by Strimzi CO, it is easily lost among them
* Very often, the restart reason is `Pod has old revision` which means that the Pod definition has changed -> but the root cause of the change could be for example updated listener certificate or something similar.

I think we should consider the future of the events used by the operator and two options for how to deal with them come to my mind:
1. We can issue them with the custom resource as the main object they reference (i.e. the `regarding` field). That would make it easier to find the events as the custom resource will have only our events and not the events related to the Pod lifecycle. The Pod might be referenced as the `related` resource if needed. Issuing the events to the custom resource might also make it easier to consider other situations when we might want to issue events.
2. We can simply remove them. There seem to be some users using them, but I think it is a relatively small number of users. So removing them might help to simplify our codebase and testing.

**Comment by scholzj** [comment]
I was thinking the Kafka, KafkaConnect, KafkaMIrrorMaker2 etc. resources

**Comment by katheris** [comment]
I would also vote for option 1 of having them use the custom resource as the main object they reference. I think it is useful to be able to see the restart as an event rather than having to always look through the operator logs.

**Comment by im-konge** [comment]
Triaged on 9.1.2025: We should go with the option 1 and keep this issue open.

**Comment by ppatierno** [comment]
> If we do this we need to consider how we represent the pod that was restarted

I guess that it was proposed to use the `related` field for the pod and the `regarding` for the `Kafka` custom resource.
Even if reading the Kubernetes API doc, it makes an example like this regarding ReplicaSet Controller:

> regarding contains the object this Event is about. In most cases it's an Object reporting controller implements, e.g. ReplicaSetController implements ReplicaSets and this event is emitted because it acts on some changes in a ReplicaSet object.

And putting `Kafka` in the `regarding` field would mean a change in the `Kafka` custom resource which is not actually true.

Maybe we could stick with `regarding` to point at the Pod and using `related` for the `Kafka` (or any other) custom resource?

**Comment by fvaleri** [comment]
Just to mention, with the UTO proposal we decided to get rid of all KafkaTopic events and no one complained. The reasoning was that most people use the resource status, operator logs and metrics.

**Comment by scholzj** [comment]
@fvaleri Yeah, I know. But for the broker pods, there are some people using them. Likely not many, but we get questions, complains, and so on about them. So it is not completely unused.

**Comment by katheris** [comment]
I've create a proposal PR to continue the discussion https://github.com/strimzi/proposals/pull/152
